### ISSUE TITLE
Installing on Ubuntu 20.04 Focal Fossa

### ISSUE THREAD

**Issue by caseyjconger** [sugarlabs/sugar#948]
Hi,

Apologies if this is answered elsewhere. But I am currently unable to install sugar on Ubuntu 20.04.

I have verified that the universe repository is enabled but when I try to install but it says that it is unable to locate the package. And just looking at https://packages.ubuntu.com/focal/allpackages I don't see it listed in the included packages.

Any insight as to how I can go about installing it on a Ubuntu 20.04 system?

Thanks

**Comment by quozl** [comment]
That's right, it was omitted from Ubuntu 20.04 because it wasn't in Debian for the few weeks that Ubuntu froze.  OLPC OS 20.04 contains Sugar packages, and they can be added to an Ubuntu 20.04 system.  Or use Ubuntu 20.10 or later.

**Comment by chimosky** [comment]
Hi @caseyjconger, were you able to fix your issue?

**Comment by caseyjconger** [comment]
Hi @chimosky Sorry, not really. I took a look and I'm not sure how to add the packages without installing the whole OLPC OS. I may just be misunderstanding though.

**Comment by quozl** [comment]
Yeah, sorry about that.  I'm about to go on night shift for a week at a telescope, so can't fix this now.  But here's some information that may help anyone else to fix it;

* OLPC OS 20.04 was a speculative product to support sales of laptops, made by one engineer,
* technical information on the product [is here](http://wiki.laptop.org/go/20.04.0),
* it was made by taking Ubuntu 20.04 and adding Sugar, which as it happened wasn't available in Ubuntu 20.04 because of aforementioned coincidences between Debian and Ubuntu operations,
* the Sugar packages were derived from the Debian packages, with some tweaks specific to OLPC.

Now Ubuntu 20.04 is old, but it is the most recent long term support release by Canonical, so there is still a reason to support it if anyone wants to do so or is being paid for it.

If someone were to make Sugar more available on Ubuntu 20.04, the shortest path is likely;
* add the OLPC 20.04 package repository at http://dev.laptop.org/~quozl/.us to sources.list,
* install the sucrose package,
* fix any regressions that occur; I think one such is that the repository release file is out of date,

Here is what olpc-ubuntu-sugar-builder package adds to sources.list on the installation image;

```
deb [trusted=yes arch=amd64] http://dev.laptop.org/~quozl/.us focal main
```

**Comment by Ark74** [comment]
Hello!
I'm helping out on the next trisquel 10.0 sugar flavor release  based on Ubuntu 20.04 (we are a bit behind from upstream).

We are backporting packages from groovy as there is no sugar release on focal, and have our own [set of activity packages](https://gitlab.trisquel.org/Ark74/trisquel-sugar/-/tree/nabia/sugar/packages) build as debian packages,

Still I'm finding issues when login, the interface seems frozen, as no activity is loaded.

```
(main.py:939): dbind-WARNING **: 22:54:41.265: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files

(metacity:994): dbind-WARNING **: 22:54:42.034: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files

(metacity-message:1013): dbind-WARNING **: 22:54:42.471: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
[...]
  File "/usr/lib/python3/dist-packages/dbus/bus.py", line 361, in get_name_owner
    return self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/usr/lib/python3/dist-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: Could not get owner of name 'org.laptop.sugar.DataStore': no such name
[...]
  File "/usr/lib/python3/dist-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.laptop.sugar.DataStore was not provided by any .service files
```

I'm kind of new to sugar, hopefully you can give me some advice, I've read that it would be best to try the latest packages available, I wonder if I should try impish instead of groovy for this release.

Thanks in advance.

**Comment by Ark74** [comment]
Well, went ahead with impish packages and it works.
Some of the activities fail, I have to look it up, but that will be another subject.
![trisquel10 0+sugar-impish](https://user-images.githubusercontent.com/7444518/139568269-7d0246dc-4d16-46ef-9511-b6e706a005cf.png)


Thank you very much.

**Comment by quozl** [comment]
Thanks, well done.  Sorry for the delay, but I recognise the problem;

> name org.laptop.sugar.DataStore was not provided by any .service files

That's caused by not installing the dependencies.  Work with your distribution packager to ensure the carquinyol or datastore component is more than just a suggestion.  A possible source of dependency information is the [Native Sugar](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md#native-sugar) section of the Development Environment instructions.

**Comment by Ark74** [comment]
Thank you, indeed I was looking at missing dependencies on my setup.

I'm adjusting everything and seems almost all activities are working but these 3,

- [measure](https://github.com/sugarlabs/Measure) - 103
- [periodictable](https://github.com/sugarlabs/periodic-table) - 2
- turtleblocks - 220

Can't find turtleblocks activity repository, any idea of where to find it?

**Comment by walterbender** [comment]
It is https://github.com/sugarlabs/turtleart-activity
